test/common/network/connection_impl_test.cc

```diff
@@ -797,35 +797,76 @@ TEST_P(ConnectionImplTest, BasicWrite) {
   disconnect(true);
 }
 
-// Validate dynamic socket options
-TEST_P(ConnectionImplTest, DynamicSocketOptions) {
+// Validate dynamic socket options receive buffer size
+TEST_P(ConnectionImplTest, DynamicSocketOptionsRecvBuffer) {
   setUpBasicConnection();
   connect();
 
   auto dynamic_sockopt_ptr = client_connection_->getDynamicSocketOptionsPtr();
-  {
-    // Test set and get socket receive buffer size
-    uint32_t set_recv_buf_size = 81920;
-    uint32_t get_recv_buf_size = 0;
-    dynamic_sockopt_ptr->setSocketRecvBufferSize(set_recv_buf_size);
-    dynamic_sockopt_ptr->getSocketRecvBufferSize(get_recv_buf_size);
+  // Test set and get socket receive buffer size
+  // Get the default buffer receive buffer size and ensure it is not equal to the value
+  // configured
+  int set_recv_buf_size = 0;
+  int get_recv_buf_size = 0;
+  int get_recv_buf_default_size = 0;
+  Api::SysCallIntResult result;
+  result = dynamic_sockopt_ptr->getSocketRecvBufferSize(get_recv_buf_default_size);
+  EXPECT_EQ(result.rc_, 0);
+  // The default receive buffer size on Linux, Mac and Windows platforms is upwards of
+  // `32768` bytes. Therefore if the default size returned is more than `32768` bytes
+  // set the value to `32768` or add `4096` bytes otherwise.
+  if (get_recv_buf_default_size > 32768) {
```

Comment:
How about just incrementing or decrementing by one to ensure the new value is not the same as the old?  Seems simpler.

Reply:
Actually, I thought about this. Say someday we run into a platform where the default is already at the max; when I increment it by 1 and apply; the system call will again set to the maximum. Then we would get get_recv_buf_default_size == get_recv_buf_size. Same is the case for minimum. I am inclined in leaving this as is. If you feel strongly about this, please let me know.